mautic/mautic
# Form field value conditions in the Campaigns

The new type of event was created for campaign - Conditions. Condition can be connected to any decision, action or another condition. This condition type can be used for more future events.

The form field value condition takes value defined in the campaign condition event and compares it with all lead's submissions of the defined form. Then it decides if the condition is true or false and triggers the next event accordingly.

This PR also fixes error for negative path of a campaigns if a decision is a first node after source.

Requested in https://github.com/mautic/mautic/issues/88, https://github.com/mautic/mautic/issues/527 and https://www.mautic.org/community/index.php/910-trigger-actions-based-on-form-fields

This also solves https://www.mautic.org/community/index.php/980-mautic-campaigns-trigger-dbalexception/0#p3905

### How to test
1. Apply the PR, clear Mautic cache, clear browser cache
2. Open a old or new campaign and create a campaign with the conditions.
3. Trigger the campaign and make sure it behaves as it should.
4. Old campaigns must still work.

Big thanks to Alan for help!
